airflow/cli/commands/task_command.py

```diff
@@ -227,6 +227,7 @@ def _run_task_by_executor(args, dag, ti):
         pickle_id=pickle_id,
         ignore_all_deps=args.ignore_all_dependencies,
         ignore_depends_on_past=args.ignore_depends_on_past,
+        wait_for_past_depends_before_skipping=args.wait_for_past_depends_before_skipping,
```

Comment:
And we can merge these two _now_ into an enum, it both is clearer and makes more sense (since as it’s designed now it is nonsensical to set _both_ flags).